src/sagemaker/cli/framework_upgrade.py

```diff
@@ -0,0 +1,284 @@
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""A Python script to upgrade framework versions"""
+from __future__ import absolute_import
+
+import argparse
+import json
+
+
+DLC_FRAMEWORK = {"tensorflow", "mxnet", "pytorch"}
+
+
+def _read_json_to_dict(filename):
+    """Read a json file into a Python dictionary
+
+    Args:
+        filename (str): Name of the json file.
+
+    Returns:
+        dict: A Python Dictionary.
+    """
+    with open(filename, "r") as f:
+        existing_content = json.load(f)
+        return existing_content
+
+
+def _get_latest_values(framework, existing_content, image_type=None):  # pylint: disable=W0621
+    """Get the latest "registries", "py_versions", "repository", values
+
+    Args:
+        framework (str): Name of the target framework.
+        existing_content (dict): Dictionary of complete framework image information.
+        image_type (str): Type of the image, required if the target is DLC
+            framework (Default: None).
+    """
+    if framework not in DLC_FRAMEWORK:
+        latest_version = list(existing_content["versions"].keys())[-1]
+        registries = existing_content["versions"][latest_version]["registries"]
+        py_versions = existing_content["versions"][latest_version][  # pylint: disable=W0621
+            "py_versions"
+        ]
+        repository = existing_content["versions"][latest_version]["repository"]
+    else:
+        if image_type is None:
+            raise ValueError(
+                "Image type ('training', 'inference', 'eia') " "is required for DLC framework."
+            )
+        latest_version = list(existing_content[image_type]["versions"].keys())[-1]
+        registries = existing_content[image_type]["versions"][latest_version]["registries"]
+        py_versions = existing_content[image_type]["versions"][latest_version]["py_versions"]
+        repository = existing_content[image_type]["versions"][latest_version]["repository"]
+
+    return registries, py_versions, repository
+
+
+def _write_dict_to_json(filename, existing_content):  # pylint: disable=W0621
+    """Write a Python dictionary to a json file.
+
+    Args:
+        filename (str): Name of the target json file.
+        existing_content (dict): Dictionary to be written to the json file.
+    """
+    with open(filename, "w") as f:
+        json.dump(existing_content, f, sort_keys=True, indent=4)
+
+
+def add_dlc_framework_version(
+    existing_content,
+    short_version,
+    full_version,
+    image_type,
+    processors,
+    py_versions,
+    registries,
+    repository,
+):  # pylint: disable=W0621
```

Comment:
what's causing the need for this disable?

Reply:
because I was reusing the name "py_versions" in methods and in __main__

Do you think we should come up with different names for the following variables?

    framework
    short_version
    full_version
    image_type
    processors
    py_versions
    scopes
    tag_prefix
    region
    account

Reply:
would the error go away if you create a `main()` method and just call that from the `if __name__ == "__main__"` section?